pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.mixed device is PennyLane's standard qubit simulator for mixed-state computations.
+
+It implements the necessary :class:`~pennylane._device.Device` methods as well as some built-in
+:mod:`qubit operations <pennylane.ops.qubit>`, providing a simple mixed-state simulation of
+qubit-based quantum circuits.
+"""
+
+import numpy as np
+
+from pennylane import QubitDevice
+from pennylane.operation import DiagonalOperation, Channel
+
+
+class DefaultMixed(QubitDevice):
```

Comment:
@trbromley raised the possibility of inheriting from `DefaultQubit` instead of `QubitDevice`. For now, it's not clear what approach is better, but we're leaning towards inheriting from `QubitDevice` to make sure the code is more explicit. Any thoughts from others?

Reply:
inheriting from `default.qubit` will effectively get you 3 simulators for the price of one (you get the "xyz.autograd" and "xyz.tf" variants effectively for free)

Reply:
Even if we overwrite basically all the methods? Because we have to overwrite basically all the methods. If so, then this is a very strong reason to inherit from `default.qubit`

Reply:
actually, checking the code a bit more carefully, this benefit is provided in `QubitDevice`

Reply:
If you had asked me 2 weeks ago, I would have also suggested inheriting from `default.qubit`. But after @trbromley's recent modifications in #772, I would now say better to inherit from `QubitDevice`!

After these changes, `default.qubit` is probably implementing too many statevector specific optimizations when applying gates.

Reply:
It seems safer to inherit from `QubitDevice` with few, if any, downsides, so let's go for that